test/e2e/specs/editor/blocks/classic.spec.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const fs = require( 'fs/promises' );
+const os = require( 'os' );
+const { v4: uuid } = require( 'uuid' );
+
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	mediaUtils: async ( { page }, use ) => {
+		await use( new MediaUtils( { page } ) );
+	},
+} );
+
+test.describe( 'Classic', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+	} );
+
+	test( 'should be inserted', async ( { editor, page, pageUtils } ) => {
+		await editor.insertBlock( { name: 'core/freeform' } );
+		// Ensure there is focus.
+		await page.click( '.mce-content-body' );
+		await page.keyboard.type( 'test' );
+		// Move focus away.
+		await pageUtils.pressKeyWithModifier( 'shift', 'Tab' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe( 'test' );
+	} );
+
+	test( 'should insert media, convert to blocks, and undo in one step', async ( {
+		editor,
+		mediaUtils,
+		page,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/freeform' } );
+		// Ensure there is focus.
+		await page.click( '.mce-content-body' );
+		await page.keyboard.type( 'test' );
+
+		await page.getByRole( 'button', { name: /Add Media/i } ).click();
+
+		const modalGalleryTab = page.getByRole( 'tab', {
+			name: 'Create gallery',
+		} );
+
+		await expect( modalGalleryTab ).toBeVisible();
+		await modalGalleryTab.click();
+
+		const filename = await mediaUtils.upload(
+			page.locator( '.media-modal .moxie-shim input[type=file]' )
+		);
+
+		// Wait for upload
+		await expect(
+			page.locator( `role=checkbox[name="${ filename }"i]` )
+		).toBeChecked();
+
+		const createGallery = page.getByRole( 'button', {
+			name: 'Create a new gallery',
+		} );
+		await expect( createGallery ).not.toBeDisabled();
+		await createGallery.click();
+		await page.click( 'role=button[name="Insert gallery"i]' );
```

Comment:
Can we also use the `getByRole` method here? 
```suggestion
		await page.getByRole( 'button' , { name: /Insert gallery/i } ).click();
```

Reply:
I think it's a matter of preference. I usually find `page.click()` simpler in similar cases.

We had a similar discussion here https://github.com/WordPress/gutenberg/pull/45193#issuecomment-1286582388.

Reply:
Thanks for the context! I can see how the `page.click()` can be simpler in some cases, although a good argument for discontinuing it is that it's not in the documentation anymore, so [it can be confusing](https://github.com/WordPress/gutenberg/pull/47498#discussion_r1089133227) for folks that don't have a lot of experience with Playwright-specific selectors. The `getBy*` locators are [the officially recommended](https://playwright.dev/docs/locators) ones now.

Reply:
cc @kevin940726

Reply:
Yeah, I just found out the other day that the role selector is not in the doc anymore. It'd be nice if we recommend using `getByRole` in our doc from now on too. I think we can also migrate all existing usages in one go during a playwright upgrade if that's feasible.

Reply:
Let's migrate all existing usages in one go during the next Playwright update.